test/chaos/src/main.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::time::Duration;
+
+use md5::{Digest, Md5};
+use rand::Rng;
+use structopt::StructOpt;
+
+use ore::retry;
+use test_util::{generator, kafka, mz_client};
+
+#[tokio::main]
+async fn main() {
+    if let Err(e) = run().await {
+        eprintln!("ERROR: {:#?}", e);
+        std::process::exit(1);
+    }
+}
+
+async fn run() -> Result<(), anyhow::Error> {
+    let timer = std::time::Instant::now();
+    let args = Args::from_args();
+    env_logger::init();
+
+    let seed: u32 = rand::thread_rng().gen();
+    let topic = format!("chaos-{}", seed);
+    log::info!(
+        "Starting chaos test with mzd={}:{} kafka={} topic={} message_count={}",
+        args.materialized_host,
+        args.materialized_port,
+        args.kafka_url,
+        topic,
+        args.message_count
+    );
+
+    // Create Kafka topic.
+    log::info!("creating Kafka topic=> {}", topic);
+    let mut kafka_client =
+        kafka::kafka_client::KafkaClient::new(&args.kafka_url, "materialize.chaos", &topic)?;
+    retry::retry_for(Duration::from_secs(10), |_| {
+        kafka_client.create_topic(args.kafka_partitions)
+    })
+    .await?;
+
+    // Generate messages and md5 hash.
+    log::info!("creating {} records", args.message_count);
+    let mut hasher = Md5::new();
+    for _ in 0..args.message_count {
+        let record = generator::bytes::generate_bytes(30);
+        hasher.input(&record);
+        kafka_client.send(&record).await?;
+    }
+    let expected_hashed = format!("{:x}", hasher.result());
```

Comment:
This is the core of it, right? Generate a bunch of data, and verify that when we do a query the hash is the same?

I'm tempted to say that this should be added as a technique inside of the new `test/checker` tool that @nacrooks  wrote, if she thinks that that would make sense? Philosophically they're slightly different, and making the external API super confusing wouldn't be helpful, but it does have tools to verify that queries return the right thing, this would just be another tool (e.g. like `check-mode = "hash-of-results"` vs what is currently there being something like `check-mode = "rows-match"`.

Reply:
To be clear this is fine as is! Just trying to make sure that we think about this as a thing that probably should be refactored going forward. @JLDLaughlin pointed out that another good (maybe better?) place for this is in the test-utils crate.

Reply:
My current dream is to start pulling all the common testing code into the `test-utils` crate (data generation, correctness checks, etc) and leaving specific testing to the other crates. I've started doing that in my past few PRs, but definitely agree that correctness should be consolidated next.